lib/cosmos/tools/cmd_tlm_server/gui/logging_tab.rb

```diff
@@ -0,0 +1,176 @@
+# encoding: ascii-8bit
+
+# Copyright 2014 Ball Aerospace & Technologies Corp.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+
+require 'cosmos'
+require 'cosmos/gui/qt'
+
+module Cosmos
+
+  # Implements the logging tab in the Command and Telemetry Server GUI
+  class LoggingTab
+
+    def initialize(production)
+      @production = production
+      @logging_layouts = {}
+    end
+
+    # Create the logging tab and add it to the tab_widget
+    #
+    # @param tab_widget [Qt::TabWidget] The tab widget to add the tab to
+    def populate(tab_widget)
```

Comment:
Convention is to have a populate(tab_widget) method and update method. The rest is private.